tests/testthat/test-codecov.R

```diff
@@ -54,396 +54,333 @@ ci_vars <- c(
 cov <- package_coverage(test_path("TestS4"))
 
 test_that("it generates a properly formatted json file", {
-  withr::with_envvar(
-    ci_vars,
-    with_mocked_bindings(
-      RETRY = function(...) list(...),
-      content = identity,
-      local_branch = function(dir) "master",
-      current_commit = function(dir) "a94a8fe5ccb19ba61c4c0873d391e987982fbbd3",
-      code = {
-        res <- codecov(coverage = cov)
-        json <- jsonlite::fromJSON(res$body)
+  local_mocked_bindings(
+    RETRY = function(...) list(...),
+    content = identity,
+    local_branch = function(dir) "master",
+    current_commit = function(dir) "a94a8fe5ccb19ba61c4c0873d391e987982fbbd3"
+  )
+  # Set environment variables
+  withr::local_envvar(ci_vars)
+  res <- codecov(coverage = cov)
+  json <- jsonlite::fromJSON(res$body)
 
-        expect_match(json$files$name, "R/TestS4.R")
-        expect_equal(
-          json$files$coverage[[1]],
-          c(
-            NA, NA, NA, NA, NA, NA, NA, 5, 2, NA, 3, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA,
-            NA, NA, NA, NA, 1, NA, NA, NA, NA, NA, 1, NA, NA, NA, NA, NA, 1, NA
-          )
-        )
-        expect_equal(json$uploader, "R")
-      }
+  expect_match(json$files$name, "R/TestS4.R")
+  expect_equal(
+    json$files$coverage[[1]],
+    c(
+      NA, NA, NA, NA, NA, NA, NA, 5, 2, NA, 3, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA,
+      NA, NA, NA, NA, 1, NA, NA, NA, NA, NA, 1, NA, NA, NA, NA, NA, 1, NA
     )
   )
+  expect_equal(json$uploader, "R")
 })
 
 test_that("it adds a flags argument to the query if specified", {
-  withr::with_envvar(
-    ci_vars,
-    with_mocked_bindings(
-      RETRY = function(...) list(...),
-      content = identity,
-      local_branch = function(dir) "master",
-      current_commit = function(dir) "a94a8fe5ccb19ba61c4c0873d391e987982fbbd3",
-      code = {
-        res <- codecov(coverage = cov, flags = "R")
-        expect_equal(res$query$flags, "R")
-      }
-    )
+  local_mocked_bindings(
+    RETRY = function(...) list(...),
+    content = identity,
+    local_branch = function(dir) "master",
+    current_commit = function(dir) "a94a8fe5ccb19ba61c4c0873d391e987982fbbd3"
   )
+  withr::local_envvar(ci_vars)
```

Comment:
I'd suggest you keep this closer to its original position to make the diff a little simpler.